app/liquid_tags/github_tag/github_readme_tag.rb

```diff
@@ -3,71 +3,79 @@
 class GithubTag
   class GithubReadmeTag
     PARTIAL = "liquids/github_readme".freeze
+    GITHUB_DOMAIN_REGEXP = %r{.*github.com/}.freeze
+    OPTION_NO_README = "no-readme".freeze
+    VALID_OPTIONS = [OPTION_NO_README].freeze
 
-    attr_reader :client, :content, :options, :readme_html
-
-    def initialize(link)
-      parsed_link = parse_link(link)
-      @options = parse_options(link)
-      @client = Octokit::Client.new(access_token: token)
-      @content = client.repository(parsed_link)
-      @readme_html = fetch_readme(parsed_link)
+    def initialize(input)
+      @repository_path, @options = parse_input(input)
     end
 
     def render
+      content = Github::Client.repository(repository_path)
+
+      if show_readme?
+        readme_html = fetch_readme(repository_path)
+      end
+
       ActionController::Base.new.render_to_string(
         partial: PARTIAL,
         locals: {
           content: content,
-          show_readme: show_readme? && readme_html.present?,
+          show_readme: readme_html.present?,
           readme_html: readme_html
         },
       )
+    rescue Github::Errors::NotFound, Github::Errors::InvalidRepository
```

Comment:
Are these going to bubble all the way up to Honeybadger? If so what action do these indicate we have to take?

Reply:
Nope,  no need, they only signal to the user that the repository they are trying to render is either not found or invalid